### PR TITLE
Feature/add pact test for sample app ocr validation

### PR DESCRIPTION
**Pull request by rkondratowicz**
Just one contract (for happy path) in this PR.
Regarding authentication, I propose going with a hardcoded s2s token - sample app will stub s2s client when verifying the contract.
Read more about this here: https://docs.pact.io/provider/handling_auth

Changes related to publishing contract will be added in next PRs.

Jira ticket: https://tools.hmcts.net/jira/browse/BPS-1279

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -314,6 +327,12 @@ dependencies {
   smokeTestImplementation sourceSets.test.runtimeClasspath
   smokeTestImplementation group: 'io.rest-assured', name: 'rest-assured'
   functionalTestImplementation group: 'com.typesafe', name: 'config', version: '1.4.0'
+
+  contractTestImplementation sourceSets.main.runtimeClasspath
+  contractTestImplementation sourceSets.test.runtimeClasspath
+  contractTestImplementation group: 'io.rest-assured', name: 'rest-assured'
+  contractTestImplementation group: 'au.com.dius', name: 'pact-jvm-consumer-java8', version: '4.0.10'
```
Comment (doncem) [inline build.gradle:334]: is this needed?

File: build.gradle [hunk 2]
```diff
@@ -56,6 +64,11 @@ test {
   failFast = true
 }
 
+task contract(type: Test, description: 'Runs the functional tests.', group: 'Verification') {
```
Comment (mdayican) [inline build.gradle:67]: description is wrong?

File: build.gradle (hunk 1 above)
Comment (rkondratowicz) [reply build.gradle]: fixed

File: build.gradle (hunk 2 above)
Comment (rkondratowicz) [reply build.gradle]: oops, fixed

**Pull request merged**